GillianCore/engine/GeneralSemantics/General/GInterpreter.ml

```diff
@@ -910,12 +939,26 @@ struct
         ]
     (* Action *)
     | LAction (x, a, es) -> (
+        Debugger_log.to_rpc
+          ~json:
+            [
+              ("x", `String x);
+              ("a", `String a);
+              ("es", `List (List.map Expr.to_yojson es));
+            ]
+          "LAction";
```

Comment:
So, by doing this, you're explicitely building some yojsons.
The whole goal of the logging in place is to have as little overhead as possible when in `-l disabled` mode, that is why every log call is captured in a `(fun fmt -> ...)`, the thing inside the `...` is not computed ever if in `-l disabled`.
Could you try to do something similar here?